web/src/main/ui/app/components/flows-new/edit-flow/ingest/ui/ingest-ui.component.ts

```diff
@@ -128,6 +134,75 @@ export class IngestUiComponent implements OnInit{
   }
 
   onChange() {
+    this.buildURIPreview();
     this.saveStep.emit(this.step);
   }
+  buildURIPreview(): void {
```

Comment:
@srinathgit @wooldridge @aebadirad On the two largest projects I've been on, we never found the investment in automated testing of the UI to be worth the potential benefits. But I'm wondering how easily we could setup automated testing for logic like this, where the output can be verified without running a web browser or looking at any HTML. This is non-trivial "presentation logic" which in theory, we could write simple JS tests for that would run very quickly. 

It might be that only 10 to 20% of the total UI code would fit into this bucket, but if testing it were easy, I think it'd be worth the investment.

Reply:
@rjrudin agreed, the new MLCP command on this page also fits into this bucket. @ayuwono, are we doing this?

Reply:
@wooldridge, for the URI preview, I usually just verify the text as we input a predetermined value. For the MLCP, I think we will verify some subset of strings from the existing scenario that we have

Reply:
To add to what Aries mentioned, both the mlcp command verification and uri preview verification can be part of unit test. I dont think they are candidates in an e2e test.